manifests/charts/istio-operator/templates/service_account.yaml

```diff
@@ -1,6 +1,5 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  namespace: {{.Values.operatorNamespace}}
```

Comment:
Don't we need namespace here?

If you run `helm3 template manifests/charts/istio-operator --namespace foo` you will get the Service, Deployment, etc do not have a namespace set, but the clusterrolebinding does.

Reply:
Declared kubernetes resources in manifest are deployed in the specified namespace. Service, Deployment, etc. are deployed in `foo` (in your example).

The namespace defined in the ClusterRoleBinding resource is to specify the subject's namespace.

However, I'm not sure if the namespace would be properly set when resources are created by `operator-init`. If not, we would need to specify namespaces in manifest resources files.

Reply:
f you run that command, it wil create a SA without a namespace set. However, the ClusterRole will reference the SA in namespace `foo`. Therefor, if you do `helm3 template manifests/charts/istio-operator --namespace foo | kubectl apply -f -` it is broken

Reply:
helm3's logic intends that if you don't specify `--namespace`, it will use the one in your context. in this way, `helm3` and `kubectl` now work in the same way. You would need to do instead: 

```
helm template manifests/charts/istio-operator | kubectl --namespace foo apply -f  -
```

Reply:
I am not sure why we would not specify the namespace in the Service/Deployment/SA but then specify it in the Deployment (env var) and ClusterRole. The only difference I can see is one is broken in half of the cases (see my command above). Furthermore, popular charts like prometheus and grafana do what I recommended: https://github.com/helm/charts/blob/master/stable/prometheus/templates/deployments/server.yaml#L13

I am not really a helm expert so I could be wrong, I just do not understand the motivation here

Reply:
To be clear I am 100% fine with removing `operatorNamespace` - just not removing the `namespace` field

Reply:
Personally, I never use the `helm` CLI without specifying the `--namespace`, so I'm also OK with you.
I only guess that not having the targeted namespace bound to the `kubectl` context is against what helm teams intended for the v3. 
Some more info in [this issue](https://github.com/helm/helm/issues/5465#issuecomment-473942223)

core maintainer @bacongobbler says:

> In general, templates should not define a namespace. This is because Helm installs objects into the namespace provided with the `--namespace` flag. By omitting this information, it also provides templates with some flexibility for post-render operations (like `helm template | kubectl create --namespace foo -f -`)

On the other hand, there's *a lot* of debate on the issue (comments are as fresh as 9 days old), people are saying that its counterintuitive to the declarative logic of helm... Anyways, if you have already made your mind, tell me and I'll add it to the templates

Reply:
I would agree with that except its broken as soon as you have explicitly references in the namespace. The alleged goal is to be able to run `helm template | kubectl create --namespace foo -f -` but it doesn't actually work.

Reply:
I have just added the namespace declaration for the Deployment, Service and ServiceAccount. It should now be alright for this part.